crates/proto/src/udp/udp_stream.rs

```diff
@@ -67,13 +68,44 @@ where
     }
 }
 
+/// Trait for UdpSocket
```

Comment:
Why do we have a separate `UdpSocket` and `DnsUdpSocket` traits?

Reply:
`UdpSocket` must implement `bind()`. However, since we use `RuntimeProvider` to create such a "connection", it only requires `send_to()` and similar methods. In other words, `DnsUdpSocket` only needs to transfer data, without caring how to establish the "connection".

Reply:
So would this also be solved by moving `RuntimeProvider` into proto?

Reply:
Nope. And this should make sense naturally, because we have `Connect` and `DnsTcpStream` counterpart for TCP in original code.